### ISSUE TITLE
HTTP auth handler complete drops POST data

### ISSUE THREAD

**Issue by VincentVanlaer** [mooltipass/extension#148]
I had the following situation occur on Firefox (110.0a1 (2023-01-10)):

1. Visit some page protected with basic auth
2. Fill in some form on that page
3. Restart the browser for some reason (updates, crash, ...)
4. Due to session restore, the page is loaded and the form can be completed further, but the basic auth credentials are gone
5. Submit the form, get redirected to the mooltipass extension page
6. Login
7. Notice that something has gone wrong, the form does not appear to be submitted and is empty
8. Try to use the back button to get back to the form, notice that the form is no longer filled (which is not how firefox normally operates, forms are restored when you use the back button)

It looks like the basic auth page does not look at the HTTP method at all and assumes that the request is a GET request. Quick fix would be to not intercept the request if the HTTP method is anything but GET. The user will then see the standard basic auth login, but at least not lose their data.

**Comment by limpkin** [comment]
is this still happening?

**Comment by VincentVanlaer** [comment]
Yes. The version of the extensions I am currently running is 1.10.17
